### ISSUE TITLE
[Feature Request]Provide rules for empty subject or body

### ISSUE THREAD

**Issue by Steppschuh** [Foundry376/Mailspring#1064]
I would like to be able to create rules that match mails with an empty subject and / or empty body:

<img width="803" alt="empty_mail" src="https://user-images.githubusercontent.com/1711417/45210835-f2013400-b290-11e8-8a13-c47a6fecbb9f.png">

**Comment by bengotow** [comment]
Hey! This seems like a great idea - I'm not sure, but I think right now this won't work because the subject mail rule does `"subject".includes(text)` , and saying text = "" isn't invalid. If we special-cased the empty string scenario, I think we could add this pretty easily!

**Comment by Phylu** [comment]
I started working on this and have the following issues:
- An empty e-mail subject is rewritten to `No Subject` (by GMail) automatically. Therefore, and rule matching for "No Subject" in the subject will work already. Not sure, how other mail providers behave here.
- When sending an empty e-mail body for test purposes, the receiving mail either contains a `<br>` (HTML email) or a `↵` newline character (text email). Therefore, I was not able to propperly test this, but it should work as expected.

I noticed, that there is a small bug in the rule processing that I will create a PR for which may resolve a small issue when comparing empty strings.
